src/types.ts

```diff
@@ -492,7 +491,7 @@ export interface StateNodeDefinition<
   TContext,
   TStateSchema extends StateSchema,
   TEvent extends EventObject
-> extends StateNodeConfig<TContext, TStateSchema, TEvent> {
+> {
```

Comment:
This wasn't really needed, all actually set properties were defined directly on StateNodeDefinition - extending StateNodeConfig only kinda ensured that StateNodeConfig is a supertype of StateNodeDefinition, but because those types are recursive and use conditional types I couldn't satisfy TS after my changes.

After thinking about it I also came to a conclusion that those types don't have to be in super-sub relationship. Definition is a processed Config, but this fact doesn't have to restrict Definition to be a subtype of Config.